src/System.Net.Sockets/src/System/Net/Sockets/BaseOverlappedAsyncResult.Windows.cs

```diff
@@ -95,14 +95,17 @@ private static unsafe void CompletionPortCallback(uint errorCode, uint numBytes,
                     }
                     else
                     {
+                        bool addRefedNativeOverlapped = false;
                         try
                         {
+                            asyncResult._nativeOverlapped.DangerousAddRef(ref addRefedNativeOverlapped);
```

Comment:
> Is that correct? Seems reasonable.

Correct

> We should add a comment explaining that we should probably get rid of SafeNativeOverlapped usage

Will add

> I think we should just use the passed in nativeOverlapped instead of the one on the asyncResult

Yup, will change.  I hadn't noticed that was being passed in to the method.